src/scripting/lua_game.cc

```diff
@@ -690,9 +693,15 @@ int LuaPlayer::hide_fields(lua_State* L) {
 	Widelands::Player& p = get(L, game);
 
 	luaL_checktype(L, 2, LUA_TTABLE);
-	const Widelands::HideOrRevealFieldMode mode = (!lua_isnone(L, 3) && luaL_checkboolean(L, 3)) ?
+	const std::string state = lua_isnone(L, 3) ? "seen" : luaL_checkstring(L, 3);
```

Comment:
I'm not happy that you change the parameter type. If there's any scenario anywhere that uses `p1:hide_fields(fields, true|false)` this will suddenly crash now. We need to keep supporting a boolean argument in v1.0 (with a deprecation notice). Check for lua_isboolean and translate true/false to the appropriate string value.